client/src/test/java/org/eclipse/hono/client/impl/TelemetrySenderImplTest.java

```diff
@@ -0,0 +1,120 @@
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *    Bosch Software Innovations GmbH - initial creation
+ */
+package org.eclipse.hono.client.impl;
+
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+import static org.mockito.Matchers.any;
+import static org.mockito.Matchers.eq;
+import static org.mockito.Mockito.*;
+
+import java.util.concurrent.atomic.AtomicReference;
+
+import org.apache.qpid.proton.amqp.messaging.Rejected;
+import org.apache.qpid.proton.message.Message;
+import org.eclipse.hono.client.MessageSender;
+import org.eclipse.hono.config.ClientConfigProperties;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import io.vertx.core.Context;
+import io.vertx.core.Future;
+import io.vertx.core.Handler;
+import io.vertx.ext.unit.TestContext;
+import io.vertx.ext.unit.junit.VertxUnitRunner;
+import io.vertx.proton.ProtonDelivery;
+import io.vertx.proton.ProtonSender;
+
+/**
+ * Tests verifying behavior of {@link TelemetrySenderImpl}.
+ *
+ */
+@RunWith(VertxUnitRunner.class)
+public class TelemetrySenderImplTest {
+
+    private Context context;
+    private ClientConfigProperties config;
+    private ProtonSender sender;
+
+    /**
+     * Sets up the fixture.
+     */
+    @SuppressWarnings("unchecked")
+    @Before
+    public void setUp() {
+
+        context = mock(Context.class);
+        doAnswer(invocation -> {
+            Handler<Void> handler = invocation.getArgumentAt(0, Handler.class);
+            handler.handle(null);
+            return null;
+        }).when(context).runOnContext(any(Handler.class));
+        config = new ClientConfigProperties();
+        sender = mock(ProtonSender.class);
+    }
+
+    /**
+     * Verifies that the sender does not wait for the peer to settle and
+     * accept a message before succeeding.
+     * 
+     * @param ctx The vert.x test context.
+     */
+    @SuppressWarnings({ "unchecked" })
+    @Test
+    public void testSendMessageDoesNotWaitForAcceptedOutcome(final TestContext ctx) {
+
+        // GIVEN a sender that has credit
+        when(sender.sendQueueFull()).thenReturn(Boolean.FALSE);
+        MessageSender messageSender = new TelemetrySenderImpl(config, sender, "tenant", "telemetry/tenant", context);
+        final AtomicReference<Handler<ProtonDelivery>> handlerRef = new AtomicReference<>();
+        doAnswer(invocation -> {
+            handlerRef.set(invocation.getArgumentAt(1, Handler.class));
+            return mock(ProtonDelivery.class);
+        }).when(sender).send(any(Message.class), any(Handler.class));
+
+        // WHEN trying to send a message
+        final Future<ProtonDelivery> result = messageSender.send("device", "some payload", "application/text", "token");
+        // which gets rejected by the peer
+        ProtonDelivery rejected = mock(ProtonDelivery.class);
```

Comment:
The mock of ProtonDelivery occurs several times - inside this class, and inside the EventSenderImplTest.

How about wrapping them into some common code? Currently not so important, but if there should be more tests in the future, it seems to be appropriate to reduce the code dups.